### ISSUE TITLE
Release a working version of Neo CLI for macOS

### ISSUE THREAD

**Issue by lock9** [neo-project/neo#3176]
**Summary or problem description**
It's not possible to run Neo CLI using OSX. This is likely caused by changes in Apple security policies.
![image](https://github.com/neo-project/neo/assets/38396062/609d795f-2067-4e7e-a2c8-267d3273b58e)
 
Another minor issue is that the latest release can only be obtained from the archived repository. Users should be able to download neo-cli from this repository.

**Do you have any solution you want to propose?**
Complete solution: Fix the build & release process to generate valid neo-cli executables.
Partial solution: Compile the project locally and make a manual release.

The problem with the second solution is that we can't be sure that the code wasn't tampered. It's also not a permanent fix.

**Comment by shargon** [comment]
>  Fix the build & release process to generate valid neo-cli executables

What is wrong?

**Comment by lock9** [comment]
It doesn't run. There is an error saying "This software needs to be updated. Contact the developer for more information"

**Comment by shargon** [comment]
should say, contact mac to find out why it says this hahaha

**Comment by kudo28** [comment]
As I got this often when running cli on MacOS. 
@lock9, I think we should somehow update the docs for the dev/user to manually allow this executable via System settings.

Here is the screenshot of those:
<img width="712" alt="Privacy   Security" src="https://github.com/neo-project/neo/assets/20470243/95740e8f-e04a-40a4-8bb0-f58ca48e802b">
<img width="711" alt="developer of neo-cli  Are you" src="https://github.com/neo-project/neo/assets/20470243/574ac3d2-7a87-472b-ae27-1ba395f09508">

This is a very normal behavior on MacOS. 
So what do you think about my approach @lock9 ?

**Comment by lock9** [comment]
Hello,

The problem is that it's not one alert. It's more like 20. Users must repeat the procedure almost indefinitely to be able to run it. Another issue is that this scares users. Nearly nobody will want to use software that 'may be malicious', even less when talking about a blockchain node and wallet.

<img width="257" alt="Screenshot 2024-03-10 at 13 30 34" src="https://github.com/neo-project/neo/assets/38396062/f34ae05b-af98-4c7e-9297-58dee5d26ef0">

One solution to 'reduce' the issue is to build and ship neo-cli as a single file. Then, the user will have to authorize it only once. It will still have the 'malicious' message problem, but that's better than being unable to run.

**Comment by vang1ong7ang** [comment]
it would be good to publish a single file for macos platforms @superboyiii

**Comment by cschuchardt88** [comment]
> it would be good to publish a single file for macos platforms @superboyiii

#3018 does just that.

**Comment by cschuchardt88** [comment]
Also PR #3103 for `neo n4 cli` will be the next release that will address all the issues.

Also any features or issues you all are more than welcome to comment at that PR.
